### ISSUE TITLE
An internal error occurred while parsing Error: EACCES: permission denied in dist-packages

### ISSUE THREAD

**Issue by vaisest** [microsoft/pylance-release#308]
## Environment data

-   Language Server version: XXX
-   OS and version: Cubbli Linux (essentially Ubuntu 20.04 LTS)
-   Python version (& distribution if applicable, e.g. Anaconda): 3.8

## Actual behaviour

The extension does not work. The first line of the python file is underlined in red, and nothing else seems to work.

## Logs

```
[Info  - 01:06:51] Pylance language server 2020.8.3 (pyright 8c5415fd) starting
[Info  - 01:06:51] Server root directory: /home/username/.vscode/extensions/ms-python.vscode-pylance-2020.8.3/server
[Info  - 01:06:51] No configuration file found.
[Info  - 01:06:51] Setting pythonPath for service "TIRA": "/usr/bin/python3"
[Error - 01:06:51] stubPath /home/username/Desktop/TIRA/typings is not a valid directory.
[Info  - 01:06:51] Assuming Python version 3.8
[Info  - 01:06:51] Assuming Python platform Linux
[Error - 01:06:51] Exception caught when installing fs watcher for:
 /usr/lib/python3.8
/usr/lib/python3.8/lib-dynload
/home/username/.local/lib/python3.8/site-packages
/usr/local/lib/python3.8/dist-packages
/usr/lib/python3/dist-packages
[Info  - 01:06:51] Searching for source files
[Info  - 01:06:51] Found 5 source files
[Info  - 01:06:51] Background analysis root directory: /home/username/.vscode/extensions/ms-python.vscode-pylance-2020.8.3/server
[Info  - 01:06:51] Background analysis started
[Error - 01:06:51] An internal error occurred while parsing /home/username/Desktop/TIRA/week1/substring.py: Error: EACCES: permission denied, scandir '/usr/local/lib/python3.8/dist-packages'
    at Object.readdirSync (fs.js:795:3)
    at p.readdirEntriesSync (/home/username/.vscode/extensions/ms-python.vscode-pylance-2020.8.3/server/pyright.bundle.js:1:449287)
    at _0x2d0654._getFilesInDirectory (/home/username/.vscode/extensions/ms-python.vscode-pylance-2020.8.3/server/pyright.bundle.js:1:100868)
    at _0x2d0654.resolveAbsoluteImport (/home/username/.vscode/extensions/ms-python.vscode-pylance-2020.8.3/server/pyright.bundle.js:1:100125)
    at _0x2d0654._resolveBestAbsoluteImport (/home/username/.vscode/extensions/ms-python.vscode-pylance-2020.8.3/server/pyright.bundle.js:1:91812)
    at _0x2d0654.resolveImport (/home/username/.vscode/extensions/ms-python.vscode-pylance-2020.8.3/server/pyright.bundle.js:1:90608)
    at t.SourceFile._resolveImports (/home/username/.vscode/extensions/ms-python.vscode-pylance-2020.8.3/server/pyright.bundle.js:1:178770)
    at /home/username/.vscode/extensions/ms-python.vscode-pylance-2020.8.3/server/pyright.bundle.js:1:172492
    at s.timeOperation (/home/username/.vscode/extensions/ms-python.vscode-pylance-2020.8.3/server/pyright.bundle.js:1:467985)
    at /home/username/.vscode/extensions/ms-python.vscode-pylance-2020.8.3/server/pyright.bundle.js:1:172380
[BG] analyzing: /home/username/Desktop/TIRA/week1/substring.py ...
[BG]   parsing: /home/username/Desktop/TIRA/week1/substring.py (30ms)
[BG]   binding: /home/username/Desktop/TIRA/week1/substring.py (2ms)
[BG]   checking: /home/username/Desktop/TIRA/week1/substring.py (2ms)
[BG] analyzing: /home/username/Desktop/TIRA/week1/substring.py (35ms)
[Error - 01:06:53] An internal error occurred while parsing /home/username/Desktop/TIRA/week1/substring.py: Error: EACCES: permission denied, scandir '/usr/local/lib/python3.8/dist-packages'
    at Object.readdirSync (fs.js:795:3)
    at Object.fs.readdirSync (electron/js2c/asar.js:605:39)
    at Object.fs.readdirSync (electron/js2c/asar.js:605:39)
    at p.readdirEntriesSync (/home/username/.vscode/extensions/ms-python.vscode-pylance-2020.8.3/server/pyright.bundle.js:1:449287)
    at _0x2d0654._getFilesInDirectory (/home/username/.vscode/extensions/ms-python.vscode-pylance-2020.8.3/server/pyright.bundle.js:1:100868)
    at _0x2d0654.resolveAbsoluteImport (/home/username/.vscode/extensions/ms-python.vscode-pylance-2020.8.3/server/pyright.bundle.js:1:100125)
    at _0x2d0654._resolveBestAbsoluteImport (/home/username/.vscode/extensions/ms-python.vscode-pylance-2020.8.3/server/pyright.bundle.js:1:91812)
    at _0x2d0654.resolveImport (/home/username/.vscode/extensions/ms-python.vscode-pylance-2020.8.3/server/pyright.bundle.js:1:90608)
    at t.SourceFile._resolveImports (/home/username/.vscode/extensions/ms-python.vscode-pylance-2020.8.3/server/pyright.bundle.js:1:178770)
    at /home/username/.vscode/extensions/ms-python.vscode-pylance-2020.8.3/server/pyright.bundle.js:1:172492
[FG] parsing: /home/username/Desktop/TIRA/week1/substring.py (22ms)
[FG] binding: /home/username/Desktop/TIRA/week1/substring.py (2ms)

```

## Code Snippet / Additional information

Happens with any Python file. Reinstall did not help. Using 2020.8.2 did not help. I am using Settings Sync beta, but it seems unlikely to be related (?). The obscure Linux version seems unlikely to be related as my Windows' WSL Ubuntu has the exact same permissions on the folder.

**Comment by heejaechang** [comment]
> Error: EACCES: permission denied, scandir '/usr/local/lib/python3.8/dist-packages' 

could we improve logging message on this?

**Comment by jakebailey** [comment]
@vaisest When Pylance is running, can you check to see which user the process is running under in top/htop? The error implies that the server can't read your files, as though it were running as the wrong user.

**Comment by vaisest** [comment]
Pylance is running on my regular user, not root. I can definitely edit the file I have open and everything else works.

Isn't that folder where the root user's installed Python scripts are? Should my user have read permissions there by default, or is Pylance at fault when it even tries to read it?

**Comment by jakebailey** [comment]
Your user should have read permissions there, as that's the global python install including the standard library. If it's not readable I would expect more to break than just us reading the files, as even running `python` from the terminal would break because it needs the standard lib.

The way that virtual environments work is by building a `sys.path` that includes the `site-packages` from your virtual environment (so you can install third party packages independently), but the standard library from the global install (so virtual environments don't copy the same thing over and over). It's normal to have to read all of these places.

**Comment by vaisest** [comment]
Seems like I made a mistake. On my system the permissions are 700 on the folder, while they are 775 on my WSL Ubuntu installation. At least the error is correct.

But anyway, there's definitely no stdlib there. The folder is empty on WSL, and on the problematic system the only thing that is there is a script I installed with `sudo pip3 install`. Python itself works correctly, and just acts like the script that's in there is not installed.
